spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -28,40 +28,60 @@
 import com.google.cloud.storage.BucketInfo;
 import com.google.cloud.storage.Storage;
 
-import org.springframework.core.io.Resource;
 import org.springframework.core.io.WritableResource;
 import org.springframework.util.Assert;
 
 /**
  * Implements {@link WritableResource} for buckets in Google Cloud Storage.
  *
  * @author Chengyuan Zhao
+ * @author João André Martins
  */
 public class GoogleStorageResourceBucket implements WritableResource {
 
 	private final Storage storage;
 
 	private final String bucketName;
 
-	private final boolean createBucketIfRequired;
+	private final boolean createBucketIfNotExists;
 
+	/**
+	 * Constructs the resource representation of a GCS bucket.
+	 * @param storage the Google Cloud Storage client
+	 * @param bucketName the name of the bucket
+	 * @param createBucketIfNotExists determines bucket auto-creation when another operation that
+	 *                                depends on the existence of this bucket is executed (e.g.,
+	 *                                creating a file in the bucket)
+	 */
 	public GoogleStorageResourceBucket(Storage storage, String bucketName,
 			boolean createBucketIfNotExists) {
 		Assert.notNull(storage, "Storage object can not be null");
 		Assert.notNull(bucketName, "Bucket name can not be null");
 
 		this.storage = storage;
 		this.bucketName = bucketName;
-		this.createBucketIfRequired = createBucketIfNotExists;
-	}
+		this.createBucketIfNotExists = createBucketIfNotExists;
+	}
+
+	/**
+	 * Creates the bucket that this {@link GoogleStorageResourceBucket} represents in Google Cloud
+	 * Storage.
+	 *
+	 * <p>If the bucket already exists in Google Cloud Storage, the
+	 * {@link com.google.cloud.storage.Bucket} object for that bucket is returned.
+	 * @return the {@link com.google.cloud.storage.Bucket} object for the bucket
+	 */
+	public Bucket create() {
+		if (exists()) {
+			throw new IllegalStateException("The bucket " + this.bucketName + " already exists.");
+		}
 
-	public void createBucket() {
-		this.storage.create(BucketInfo.newBuilder(this.bucketName).build());
+		return this.storage.create(BucketInfo.newBuilder(this.bucketName).build());
```

Comment:
If the bucket already exists, wouldn't this throw some kind of an exception? If so, we can probably remove the if statement above and avoid an extra existence check.